examples/nvexec/reduce.cpp

```diff
@@ -1,8 +1,6 @@
 #include <nvexec/stream_context.cuh>
 #include <stdexec/execution.hpp>
 
-// CUB relies on __reduce_add_sync which Clang CUDA doesn't provide
-#if !STDEXEC_CLANG()
```

Comment:
This doesn't work for me. I'm using clang++16:
```bash
/usr/bin/clang++-16   -I/home/gevtushenko/src/senior-zero/stdexec/examples/nvexec -I/home/gevtushenko/src/senior-zero/stdexec/include -I/home/gevtushenko/src/senior-zero/stdexec/build/_deps/libcudacxx-src/lib/cmake/libcudacxx/../../../include -w -O3 -DNDEBUG --cuda-gpu-arch=sm_86 --cuda-path=/usr/local/cuda -Wall -ferror-limit=0 -fmacro-backtrace-limit=0 -ftemplate-backtrace-limit=0 -fcolor-diagnostics -Wno-unknown-cuda-version -Xclang=-fcuda-allow-variadic-functions -std=c++20 -MD -MT examples/nvexec/CMakeFiles/reduce.dir/reduce.cpp.o -MF examples/nvexec/CMakeFiles/reduce.dir/reduce.cpp.o.d -x cuda -c /home/gevtushenko/src/senior-zero/stdexec/examples/nvexec/reduce.cpp -o examples/nvexec/CMakeFiles/reduce.dir/reduce.cpp.o
```

You must be compiling with older version of CTK, which doesn't contain ampere reduction support.